compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/DocumentationParser.java

```diff
@@ -266,10 +378,81 @@ private STNode parseBacktickToken() {
     }
 
     /**
-     * Parse back-tick content token.
+     * Parse back-tick content.
      *
+     * @param referenceType Node that precedes the backtick block
      * @return Parsed node
      */
+    private STNode parseBacktickContent(STNode referenceType) {
+        ReferenceGenre referenceGenre = getReferenceGenre(referenceType);
+        if (isValidBacktickContentSequence(referenceGenre)) {
+            return parseBacktickContent();
+        }
+
+        STNode contentToken = combineAndCreateBacktickContentToken();
+        if (referenceGenre != ReferenceGenre.NO_KEY) {
+            // Log warning for backtick block with a reference type, but content is invalid.
+            contentToken = SyntaxErrors.addDiagnostic(contentToken,
+                    DiagnosticWarningCode.WARNING_INVALID_DOCUMENTATION_IDENTIFIER, ((STToken) contentToken).text());
+        }
```

Comment:
Don't we give a warning for invalid backtick content with `NO_KEY`. Also `combineAndCreateBacktickContentToken` can be renamed to `combineAndCreateInvalidBacktickContentToken`

Reply:
We need to clarify it with the spec. There were several reference like above in repo atm. Create an issue for that. #27958
+1 on changing the name. Will do once clarified.